x-pack/plugins/serverless/server/plugin.ts

```diff
@@ -63,7 +63,7 @@ export class ServerlessPlugin implements Plugin<ServerlessPluginSetup, Serverles
             // with a specific config.  So in this case, to ensure the switcher remains enabled,
             // write the selected config to `recent` and tack on the setting to enable the switcher.
             writeFileSync(
-              resolve(getConfigDirectory(), 'serverless.recent.yml'),
```

Comment:
@elastic/appex-sharedux  - could you check if you're okay with this small change. We changed the name of the generated config file to `serverless.recent.dev.yml`